src/loader/bgw_launcher.c

```diff
@@ -258,7 +263,8 @@ populate_database_htab(void)
 	 * first initialize the hashtable before we start a txn, we'll be writing
 	 * info about dbs there
 	 */
-	db_htab = hash_create("launcher_db_htab", 8, &info, HASH_BLOBS | HASH_ELEM);
+	staging_db_htab = hash_create("staging_db_htab", guc_max_background_workers, &info, HASH_BLOBS | HASH_ELEM);
```

Comment:
No reason for hash table. Changed to list.

I didn't separate it out into another function because it looks like lists, as opposed to these shmem hash tables, are allocated in the current memory context. `populate_database_htab` is running in a transaction, so the list objects are going to be allocated in the transactional context, which goes away if we call `CommitTransaction` (which would be at the end of the first function if we broke the logic up). We could get around this by switching memory contexts everytime we append to the list, but I feel that isn't worth the refactoring? I think it's ok as is?